### PR TITLE
pythonPackages.karton-core: init at 4.2.0

### PR DESCRIPTION
**Pull request by chivay**
###### Motivation for this change

Karton is a Python-based framework designed for building malware analysis pipelines.

Add myself as the maintainer.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119082` at 06982f83 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.karton-core</li>
    <li>python39Packages.karton-core</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Related to #81418

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1707,6 +1707,12 @@
     githubId = 3086255;
     name = "Barry Moore II";
   };
+  chivay = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:1710]: This change should go into a separate commit in this PR.

Comment (chivay) [reply maintainers/maintainer-list.nix:1710]: Fixed

File: pkgs/development/python-modules/karton-core/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchFromGitHub {
+    owner = "CERT-Polska";
+    repo = "karton";
+    rev = "v${version}";
+    sha256 = "08j1bm9g58576sswcrpfczaki24nlqqaypp7qv1rxxwsyp5pq6h6";
+  };
+
+  propagatedBuildInputs = [ minio redis ];
+
+  checkPhase = "python -m unittest discover";
```
Comment (fabaff) [inline pkgs/development/python-modules/karton-core/default.nix:21]:
```suggestion
  checkPhase = ''
    runHook preCheck
    ${python.interpreter} -m unittest discover
    runHook postCheck
  '';
```

and add `python` to the inputs.

File: pkgs/development/python-modules/karton-core/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    description = "Distributed malware processing framework based on Python, Redis and MinIO";
```
Comment (fabaff) [inline pkgs/development/python-modules/karton-core/default.nix:24]: I suggest to keep it simple.

```suggestion
    description = "Distributed malware processing framework";
```

File: pkgs/development/python-modules/karton-core/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    homepage = "https://github.com/CERT-Polska/karton";
```
Comment (fabaff) [inline pkgs/development/python-modules/karton-core/default.nix:25]: One can get to the documentation from the source repo with one click but from my point of view it's nicer to link directly to the documentation.

```suggestion
    homepage = "https://karton-core.readthedocs.io/";
```

**Comment by chivay** [comment]
Applied all suggestions

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 119082` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.karton-core</li>
    <li>python39Packages.karton-core</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package built (see https://github.com/NixOS/nixpkgs/pull/119082#issuecomment-817318897)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-119082/results/python38Packages.karton-core/lib/python3.8/site-packages]$ python
Python 3.8.8 (default, Feb 19 2021, 11:04:50) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from karton.core import Producer, Task
>>> if __name__ == "__main__":
...     foo_producer = Producer(identity="foobar-producer")
...     for i in range(5):
...         task = Task(headers={"type": "foobar"}, payload={"n": i})
...         foo_producer.send_task(task)
... 
Traceback (most recent call last):
[...]
  File "/nix/store/z7hcf0c5s91nynci7xn61l35jkgp75z8-python3.8-karton-core-4.2.0/lib/python3.8/site-packages/karton/core/config.py", line 43, in __init__
    raise RuntimeError("Missing MinIO configuration")
RuntimeError: Missing MinIO configuration
```

Looks good to me.

**Comment by chivay** [comment]
Thanks!
